modules/hal_silabs/simplicity_sdk/CMakeLists.txt

```diff
@@ -77,8 +99,17 @@ if(CONFIG_SOC_GECKO_HAS_RADIO)
     zephyr_library_sources(${RADIO_DIR}/rail_lib/plugin/pa-conversions/pa_curves_efr32.c)
     zephyr_library_sources(${RADIO_DIR}/rail_lib/plugin/pa-conversions/pa_conversions_efr32.c)
 
-    # prebuilt libs
-    add_prebuilt_library(librail platform/radio/rail_lib/autogen/librail_release/librail_efr32xg${SILABS_DEVICE_FAMILY_NUMBER}_gcc_release.a)
+    # RAIL lib and config for modules
+    if("${CONFIG_SOC_PART_NUMBER}" STREQUAL "MGM240SD22VNA")
+      add_prebuilt_library(librail platform/radio/rail_lib/autogen/librail_release/librail_module_efr32xg24_gcc_release.a)
+      add_prebuilt_library_with_whole_archive(librail_config platform/radio/rail_lib/autogen/librail_release/librail_config_mgm240sd22vna_gcc.a)
```

Comment:
This should generically use the `CONFIG_SOC` variable to produce the library path.